### PR TITLE
[BTS-1259][DEVEL] Gharial CRUD Vertex / Edge operations now always do check the GraphDefinition

### PR DESCRIPTION
**Pull request by hkernbach**
### Scope & Purpose

This PR now added checks for all gharial based CRUD operations on all vertex and on all edge operations like:
* create edge (but collection is not part of the graph definition) ...
* drop edge (but collection is not part of the graph) ...
* same for vertex operations
* ... 

- [x] :hankey: Bugfix

### Checklist

- [x] Tests
  - [x] **integration tests**
- [x] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [x] Backports
  - [ ] Backport for 3.11: *(Please link PR - Yes we should backport)*
  - [ ] Backport for 3.10: *(Please link PR - Yes we should backport)*

#### Related Information

- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/BTS-1259

### CODE REVIEW

**Comment by hkernbach** [comment]
Jenkins: https://jenkins01.arangodb.biz/job/arangodb-matrix-pr/31433/

File: tests/js/client/shell/api/multi/general-graph.js [hunk 1]
```diff
@@ -1257,179 +1298,195 @@ function check400CRUD (doc) {
...
+function check404_graph_not_found(doc) {
   assertEqual(doc.code, internal.errors.ERROR_HTTP_NOT_FOUND.code);
   assertTrue(doc.parsedBody['error']);
   assertEqual(doc.parsedBody['code'], internal.errors.ERROR_HTTP_NOT_FOUND.code);
   assertEqual(doc.parsedBody['errorNum'], internal.errors.ERROR_GRAPH_NOT_FOUND.code);
   assertEqual(doc.parsedBody['errorMessage'], "graph 'UnitTestUnknown' not found");
+}
+
+const assertVertexCollectionNotKnown = (response) => {
+  assertTrue(response.parsedBody);
```
Comment (jsteemann) [inline tests/js/client/shell/api/multi/general-graph.js:1342]: This assertion does not make sense to me as a reader of the code. Did you mean `assertTrue(response.hasOwnProperty('parsedBody'));`?

File: tests/js/client/shell/api/multi/general-graph.js [hunk 2]
```diff
@@ -1257,179 +1298,195 @@ function check400CRUD (doc) {
...
+  assertTrue(response.parsedBody.error);
+  assertEqual(
+    response.parsedBody.errorNum,
+    internal.errors.ERROR_GRAPH_REFERENCED_VERTEX_COLLECTION_NOT_USED.code
+  );
+};
+
+const assertEdgeCollectionNotKnown = (response) => {
+  assertTrue(response.parsedBody);
```
Comment (jsteemann) [inline tests/js/client/shell/api/multi/general-graph.js:1351]: same

File: tests/js/client/shell/api/multi/general-graph.js (hunk 1 above)
Comment (hkernbach) [reply tests/js/client/shell/api/multi/general-graph.js:1342]: Ah, yes - better! Thanks for spotting. This has been just a copy-paste. Will replace all occurrences with your suggestion.

File: tests/js/client/shell/api/multi/general-graph.js (hunk 2 above)
Comment (hkernbach) [reply tests/js/client/shell/api/multi/general-graph.js:1351]: Ah, yes - better! Thanks for spotting. This has been just a copy-paste. Will replace all occurrences with your suggestion.

**Review by jsteemann — APPROVED** [review]
LGTM

File: tests/js/client/shell/api/multi/general-graph.js (hunk 2 above)
Comment (Tasya201) [reply tests/js/client/shell/api/multi/general-graph.js]: yes

**Pull request merged**